src/run_nps_wbm.R

```diff
@@ -0,0 +1,680 @@
+    d.r = 1 + 0.033*cos((2*pi/365)*doy)
+    declin = 0.409*sin((((2*pi)/365)*doy)-1.39)
+    lat.rad = (pi/180)*lat
+    sunset.ang = acos(-tan(lat.rad)*tan(declin))
+    R.a = ((24*60)/pi)*0.082*d.r*((sunset.ang*sin(lat.rad)*sin(declin)) + (cos(lat.rad)*cos(declin)*sin(sunset.ang)))
+    Oudin = ifelse(snowpack>2,0,ifelse(tmean>-5,(R.a*(tmean+5)*0.408)/100,0))
+    Folded_aspect = abs(180-abs((aspect)-225))
+    Heatload = (0.339+0.808*cos(lat*(pi/180))*cos(slope*(pi/180)))-(0.196*sin(lat.rad)*sin(slope*(pi/180)))-(0.482*cos(Folded_aspect*(pi/180))*sin(slope*(pi/180)))
+    sc = ifelse(!is.null(shade.coeff), shade.coeff, 1)
+    OudinPET = Oudin * Heatload * sc
+    return(OudinPET)
+  }
+
+
+
+  # Now, run WBM
+
+  print("Running wbm...")
+
+  calculator <- function(pt_nums){
+    start = Sys.time()
+    #for (i in 1:nrow(points2)) {
+
+    point <- data.table(points2) %>% dplyr::filter(pt_num == pt_nums)
+    
+    DailyWB <- data.table(climate_data) %>% dplyr::filter(pt_num == pt_nums)
+    DailyWB$doy <- yday(DailyWB$date)
+    DailyWB$daylength <- get_daylength(DailyWB$date, point$y)
+    DailyWB$jtemp = as.numeric(get_jtemp(Lon = point$x, Lat = point$y))
+    DailyWB$F = get_freeze(DailyWB$jtemp, DailyWB$tmean_C)
+    DailyWB$RAIN = get_rain(DailyWB$ppt_mm, DailyWB$F)
+    DailyWB$SNOW = get_snow(DailyWB$ppt_mm, DailyWB$F)
+    DailyWB$MELT = get_melt(DailyWB$tmean_C, DailyWB$jtemp, hock=hock_coef, DailyWB$SNOW, point$Snowpack.Init)
+    DailyWB$PACK = get_snowpack(DailyWB$jtemp, DailyWB$SNOW, DailyWB$MELT)
+    DailyWB$W = DailyWB$MELT + DailyWB$RAIN
+    if(PET_Method == "Hamon"){
+      DailyWB$PET = ET_Hamon_daily(DailyWB)
+    } else {
+      if(PET_Method == "Penman-Monteith"){
+        DailyWB$PET = ET_PenmanMonteith_daily(DailyWB)
+      } else {
+        if(PET_Method == "Oudin"){
+          DailyWB$PET = get_OudinPET(DailyWB$doy, point$y, DailyWB$PACK,
+                                     DailyWB$tmean_C, point$Slope,
+                                     point$Aspect, point$Shade.Coeff)
+        } else {
+          print("Error - PET method not found")
+        }
+      }
+    }
+    #DailyWB$PET_mod = modify_PET(DailyWB$PET, DailyWB$Slope, DailyWB$Aspect,
+    #                         DailyWB$Lat, DailyWB$Shade.Coeff)
+    DailyWB$PET_mod <- DailyWB$PET
+    DailyWB$W_PET = DailyWB$W - DailyWB$PET_mod
+    DailyWB$SOIL = get_soil(DailyWB$W, point$Soil.Init, DailyWB$PET_mod,
+                            DailyWB$W_PET, point$SWC.Max)
+    DailyWB$DSOIL = diff(c(point$Soil.Init, DailyWB$SOIL))
+    DailyWB$AET = get_AET(DailyWB$W, DailyWB$PET_mod, DailyWB$SOIL, point$Soil.Init)
+    DailyWB$W_ET_DSOIL = DailyWB$W - DailyWB$AET - DailyWB$DSOIL
+    DailyWB$D = DailyWB$PET_mod - DailyWB$AET
+    DailyWB$GDD = get_GDD(DailyWB$tmean_C, T.Base)
+
+    DailyWB_ret <- DailyWB %>%
+      data.table() %>%
+      ungroup() %>%
+      dplyr::select(date, x, y, pt_num, PR = pr, RUNOFF = W_ET_DSOIL, RAIN, SNOW, MELT) %>%
```

Comment:
Change "pr" to "ppt_mm" because we're getting rid of redundant columns upstream so this won't be available (they are the same thing here so inconsequential).